### PR TITLE
LLVM and SPIRV-LLVM-Translator pulldown (WW42)

### PR DESCRIPTION
**Pull request by sys-ce-bb**
LLVM: llvm/llvm-project@8505c3b15bfc
SPIRV-LLVM-Translator: KhronosGroup/SPIRV-LLVM-Translator@d009cd45ad56b162796a7662cc2b5b570b21ab3d

### CODE REVIEW

File: llvm/lib/SYCLLowerIR/CompileTimePropertiesPass.cpp [hunk 1]
```diff
@@ -36,7 +36,7 @@ constexpr StringRef SPIRV_PARAM_DECOR_MD_KIND = "spirv.ParameterDecorations";
 // The corresponding SPIR-V OpCode for the host_access property is documented
 // in the SPV_INTEL_global_variable_decorations design document:
 // https://github.com/intel/llvm/blob/sycl/sycl/doc/extensions/DeviceGlobal/SPV_INTEL_global_variable_decorations.asciidoc#decoration
```
Comment (vmaksimo) [inline llvm/lib/SYCLLowerIR/CompileTimePropertiesPass.cpp:38]: "SPV_INTEL_global_variable_host_access" and https://github.com/KhronosGroup/SPIRV-Registry/blob/main/extensions/INTEL/SPV_INTEL_global_variable_host_access.asciidoc#decoration

**Review by vmaksimo — COMMENTED** [review]
Review for adapting global variables SPIRV decorations change https://github.com/intel/llvm/pull/11601/commits/e034f08475d63f942cda52e54aa9cf55adc69d7f#

File: llvm/lib/SYCLLowerIR/CompileTimeProperties.def [hunk 2]
```diff
@@ -15,8 +15,8 @@
 // sycl-implement-in-csr properties are documented in the
 // SPV_INTEL_global_variable_decorations design document:
 // https://github.com/intel/llvm/blob/sycl/sycl/doc/extensions/DeviceGlobal/SPV_INTEL_global_variable_decorations.asciidoc#decoration
```
[the remark this answers is not in the document]
Comment (jsji) [reply llvm/lib/SYCLLowerIR/CompileTimeProperties.def]: Updated.

File: llvm/lib/SYCLLowerIR/CompileTimePropertiesPass.cpp (hunk 1 above)
Comment (jsji) [reply llvm/lib/SYCLLowerIR/CompileTimePropertiesPass.cpp]: Updated.

File: llvm/lib/SYCLLowerIR/CompileTimePropertiesPass.cpp [hunk 3]
```diff
@@ -36,7 +36,7 @@ constexpr StringRef SPIRV_PARAM_DECOR_MD_KIND = "spirv.ParameterDecorations";
...
-constexpr uint32_t SPIRV_HOST_ACCESS_DECOR = 6147;
+constexpr uint32_t SPIRV_HOST_ACCESS_DECOR = 6168;
 constexpr uint32_t SPIRV_HOST_ACCESS_DEFAULT_VALUE = 2; // Read/Write
```
[the remark this answers is not in the document]
Comment (jsji) [reply llvm/lib/SYCLLowerIR/CompileTimePropertiesPass.cpp]: Update.

File: sycl/doc/design/spirv-extensions/SPV_INTEL_global_variable_decorations.asciidoc
[the remark this answers is not in the document]
Comment (jsji) [reply sycl/doc/design/spirv-extensions/SPV_INTEL_global_variable_decorations.asciidoc:1]: Undo the changes.

**Review by jsji — COMMENTED** [review]
Thanks, updated.

**Comment by jsji** [comment]
>  SPIRV module requires unsupported capability 6167

We have some failures in intel GPU tests. 
Looks like we will need OCL runtime update as well, so I am going to revert to not picking up the global variables SPIRV decorations change for now.

**Comment by jsji** [comment]
This is ready for review. @bader  @intel/llvm-gatekeepers 
- [ ] [Fix warning in LangRef.rst](https://github.com/intel/llvm/pull/11601/commits/6da29c5472dcc5555376653795ba7248e2a2218e) Minor fix to doc build.
- [ ] [[SYCL][TEST] Update function-attrs for windows](https://github.com/intel/llvm/pull/11601/commits/73adcec16a5534508699c86673f933f7c67bc206) Minor test update to accommodate windows path.

**Review by bb-sycl — APPROVED** [review]

**Comment by bb-sycl** [comment]
Fri 20 Oct 2023 09:51:32 PM UTC --- Start to merge the commit into sycl branch. It will take several minutes.

**Comment by bader** [comment]
> * [ ]  [Fix warning in LangRef.rst](https://github.com/intel/llvm/pull/11601/commits/6da29c5472dcc5555376653795ba7248e2a2218e) Minor fix to doc build.

This change is not related to SYCL, so, please, commit this change to [llvm-project ](https://github.com/llvm/llvm-project/).

**Comment by bb-sycl** [comment]
Fri 20 Oct 2023 09:56:06 PM UTC --- Merge the branch in this PR to base automatically. Will close the PR later.

**Comment by jsji** [comment]
> > * [ ]   [Fix warning in LangRef.rst](https://github.com/intel/llvm/pull/11601/commits/6da29c5472dcc5555376653795ba7248e2a2218e) Minor fix to doc build.
> 
> This change is not related to SYCL, so, please, commit this change to [llvm-project ](https://github.com/llvm/llvm-project/).

This is changes `sycl` branch specific, the change was introduced in https://github.com/intel/llvm/pull/8134/files#diff-1d1772820f1edf0653335111c7ab690c53a2df20887eadadf55d7a9c17aa907a

**Pull request merged**